### PR TITLE
Changes to HiRep-CUDA

### PR DESCRIPTION
**Pull request by erikkjellgren**

### CODE REVIEW

File: Include/cluster_info.h [hunk 1]
```diff
@@ -0,0 +1,93 @@
+/**
+ * @file cluster_info.h
+ * @brief Functions that print collective information on cluster 
+ *        drivers, necessary software and available hardware.
+ *        To be logged to simulation outfiles.
+ */
+
+#ifndef CLUSTER_INFO_H
```
Comment (claudiopica) [inline Include/cluster_info.h:8]: The name is misleading. This file should be named gpu_info.h

File: Include/cross_compilation.h [hunk 2]
```diff
@@ -0,0 +1,30 @@
+/**
+ * @file cross_compilation.h
```
Comment (claudiopica) [inline Include/cross_compilation.h:2]: This is not a good idea. Remove the file.

File: Include/error.h [hunk 3]
```diff
@@ -3,34 +3,110 @@
...
+ * @file error.h
+ * @brief Functions and macros for error handling.
+ */
 
 #ifndef ERROR_H
 #define ERROR_H
 
-#ifdef __cplusplus
-extern "C" {
+#include "cross_compilation.h"
```
Comment (claudiopica) [inline Include/error.h:14]: remove

File: Include/new_geometry.h [hunk 4]
```diff
@@ -0,0 +1,40 @@
+/**
+ * @file new_geometry.h
+ * @brief Functions needed for the new geometry implementation that
+ *        will replace the current geometry in the future
+ */
+
+ #ifndef NEW_GEOMETRY_H
+ #define NEW_GEOMETRY_H
+
+static inline int lexi(int b0, int b1, int b2, int b3, int x0, int x1, int x2, int x3);
```
Comment (claudiopica) [inline Include/new_geometry.h:10]: static functions must not be included in the header file.
From this file we only need the 5 functions at the bottom after the stddef include.

**Review by claudiopica — CHANGES_REQUESTED** [review]
Each header file should have its own include guard.

**Comment by claudiopica** [comment]
If you prefer I can merge first and then you can make the changes afterwards.

**Comment by erikkjellgren** [comment]
Do not merge yet. I am not sure yet, if I have not broken something in the conflict-resolution

**Pull request merged**